### PR TITLE
Add IndicQA dataset in Retrieval task

### PR DESCRIPTION
**Pull request by jaygala24**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good! Feel free to add points

**Comment by jaygala24** [comment]
**Points Summary**:

This is the first dataset spanning 11 Indic languages in the Retrieval task category. The total points would be 2 + 4 x 11 = 46 points.

I have added the points for dataset contribution and PR review.

@KennethEnevoldsen I am also interested in reviewing PRs if possible. I am curious to understand the criteria to be a reviewer for this repository.

File: mteb/tasks/Retrieval/multilingual/IndicQARetrieval.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        dialect=[],
+        text_creation="machine-translated and verified",
+        bibtex_citation="""@article{doddapaneni2022towards,
+  title     = {Towards Leaving No Indic Language Behind: Building Monolingual Corpora, Benchmark and Models for Indic Languages},
+  author    = {Sumanth Doddapaneni and Rahul Aralikatte and Gowtham Ramesh and Shreyansh Goyal and Mitesh M. Khapra and Anoop Kunchukuttan and Pratyush Kumar},
+  journal   = {Annual Meeting of the Association for Computational Linguistics},
+  year      = {2022},
+  doi       = {10.18653/v1/2023.acl-long.693}
+}""",
+        n_samples={"test": 1700},
```
[the remark this answers is not in the document]
Comment (jaygala24) [reply mteb/tasks/Retrieval/multilingual/IndicQARetrieval.py]: I have limited the split size to 1500 for each of the languages.

File: docs/mmteb/points/547.jsonl [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{"GitHub": "jaygala24", "New dataset": 40}
+{"GitHub": "PranjalChitale", "New dataset": 16}
```
[the remark this answers is not in the document]
Comment (jaygala24) [reply docs/mmteb/points/547.jsonl]: Thanks for pointing this out. I somehow missed this. I have updated the points as well as the above message.

File: mteb/tasks/Retrieval/multilingual/IndicQARetrieval.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply mteb/tasks/Retrieval/multilingual/IndicQARetrieval.py]: This should be the total size.

**Comment by KennethEnevoldsen** [comment]
Wonderful @jaygala24, will merge this in!

**Pull request merged**